internal/testutils/resolver.go

```diff
@@ -0,0 +1,79 @@
+/*
+ *
+ * Copyright 2023 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package testutils
+
+import (
+	"testing"
+
+	"google.golang.org/grpc/internal"
+	"google.golang.org/grpc/internal/pretty"
+	"google.golang.org/grpc/resolver"
+	"google.golang.org/grpc/serviceconfig"
+)
+
+// ResolverClientConn is a fake implemetation of the resolver.ClientConn
+// interface to be used in tests.
+type ResolverClientConn struct {
+	resolver.ClientConn // Embedding the interface to avoid implementing deprecated methods.
```

Comment:
Hmm ... I don't understand this comment.

Reply:
Writing something like `var _ = ResolverClientConn{}.NewAddress` results in an init time panic since this method is not implemented in `ResolverClientConn`. I initially made a commit which had this and failed to realize that it was causing a panic. So, I had to undo this in the latest commit.